python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -0,0 +1,354 @@
+        CompareWarpAssembleCoaddConfig.setDefaults(self)
+
+
+class DcrAssembleCoaddTask(CompareWarpAssembleCoaddTask):
+
+    ConfigClass = DcrAssembleCoaddConfig
+    _DefaultName = "dcrAssembleCoadd"
+
+    def __init__(self, *args, **kwargs):
+        """!
+        \brief Initialize the task and make the \ref AssembleCoadd_ "assembleStaticSkyModel" subtask.
+        """
+        CompareWarpAssembleCoaddTask.__init__(self, *args, **kwargs)
+
+    @pipeBase.timeMethod
+    def run(self, dataRef, selectDataList=[]):
+        """!
+        \brief Assemble a coadd from a set of Warps
+
+        Coadd a set of Warps. Compute weights to be applied to each Warp and find scalings to
+        match the photometric zeropoint to a reference Warp. Optionally, match backgrounds across
+        Warps if the background has not already been removed. Assemble the Warps using
+        \ref assemble. Interpolate over NaNs and optionally write the coadd to disk. Return the coadded
+        exposure.
+
+        \anchor runParams
+        \param[in] dataRef: Data reference defining the patch for coaddition and the reference Warp
+                        (if config.autoReference=False). Used to access the following data products:
+                        - [in] self.config.coaddName + "Coadd_skyMap"
+                        - [in] self.config.coaddName + "Coadd_ + <warpType> + "Warp" (optionally)
+                        - [out] self.config.coaddName + "Coadd"
+        \param[in] selectDataList[in]: List of data references to Warps. Data to be coadded will be
+                                   selected from this list based on overlap with the patch defined by dataRef.
+
+        \return a pipeBase.Struct with fields:
+                 - coaddExposure: coadded exposure
+                 - nImage: exposure count image
+        """
+        skyInfo = self.getSkyInfo(dataRef)
+        calExpRefList = self.selectExposures(dataRef, skyInfo, selectDataList=selectDataList)
+        if len(calExpRefList) == 0:
+            self.log.warn("No exposures to coadd")
+            return
+        self.log.info("Coadding %d exposures", len(calExpRefList))
+
+        tempExpRefList = self.getTempExpRefList(dataRef, calExpRefList)
+        inputData = self.prepareInputs(tempExpRefList)
+        self.log.info("Found %d %s", len(inputData.tempExpRefList),
+                      self.getTempExpDatasetName(self.warpType))
+        if len(inputData.tempExpRefList) == 0:
+            self.log.warn("No coadd temporary exposures found")
+            return
+
+        supplementaryData = self.makeSupplementaryData(dataRef, selectDataList)
+        # nImage is created if it is requested by self.config.  Otherwise, None.
+
+        retStructGen = self.assemble(
+            skyInfo, inputData.tempExpRefList, inputData.imageScalerList, inputData.weightList,
+            supplementaryData=supplementaryData,
+        )
+        coaddExp = None
+        for retStruct in retStructGen:
```

Comment:
I think our ability to make a generic `run()` for both `dcrAssemble` and the other classes depends on how others feel about always iterating over the outputs of `assemble`.  In most cases there will be one.  It's fine with me. @r-owen? @TallJimbo?